src/ripple/rpc/impl/ShardArchiveHandler.cpp

```diff
@@ -146,93 +311,105 @@ ShardArchiveHandler::start()
 Json::Value
 ShardArchiveHandler::pause()
 {
+    std::lock_guard<std::mutex> lock(m_);
     assert(downloader_);
 
     auto response = downloader_->pause();
 
     using T = SSLHTTPDownloader::PauseResponse;
     Json::Value ret;
 
+    bool persist = false;
+
     switch(response)
     {
         case T::success:
-            ret = RPC::makeObjectValue("Shard downloader paused.");
-            break;
-        case T::no_downloads:
-            ret = make_error(rpcNO_DOWNLOADS,
-                "There are no downloads to pause.");
+            ret = makeObjectValue("Shard downloader paused.");
+            persist = true;
             break;
         case T::already_paused:
             ret = make_error(rpcALREADY_PAUSED,
                 "The downloader is already paused.");
             break;
     }
 
+    if (persist)
+    {
+        auto db = conn_->checkoutDb();
+
+        *db << "UPDATE MetaState SET Value=1 WHERE Key=\"paused\";";
+    }
+
     return ret;
 }
 
 Json::Value
 ShardArchiveHandler::resume()
 {
+    std::unique_lock<std::mutex> lock(m_);
     assert(downloader_);
 
+    auto doPersist = [this]
+    {
+        auto db = conn_->checkoutDb();
```

Comment:
No need for a LockedSociSession. conn_ is always used while holding the lock, getSession() can be used instead.